app/controllers/concerns/report_embeddable.rb

```diff
@@ -32,7 +33,10 @@ def build_report_data(options = {})
   # returns true if no errors, false otherwise
   def run_and_handle_errors
     begin
-      @report.run(current_ability)
+      Rails.cache.fetch(@report.cache_key_with_responses(current_mission)) do
+        @report.run(current_ability)
+        @report
+      end
```

Comment:
I don't think that's necessary. The block should return @report due to line 38. Or if you want to change run to return self that's fine too. We could then get rid of line 38.